src/components/deviceMap/deviceMap.js

```diff
@@ -49,7 +49,12 @@ class DeviceMap extends Component {
     this.addScript(nextProps);
     if (nextProps.BingMapKey && nextProps.BingMapKey !== config.STATUS_CODES.STATIC) {
        this.showMap(nextProps);
-     }
+    }
+    if (!nextProps.alarmList.length && !nextProps.devices.length && !nextProps.telemetryByDeviceGroup.Items.length ) {
```

Comment:
You can simplify the `if` since the `if` condition is the result you want to save in the state.
```
const { alarmList, devices, telemetryByDeviceGroup } = nextProps;
const showNoDataOverlay = !alarmList.length && !devices.length && !telemetryByDeviceGroup.Items.length;
this.setState({ showNoDataOverlay });
```